core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/PythonUDFOperator.java

```diff
@@ -0,0 +1,700 @@
+                case DATETIME:
+                    StructVector dateTimeStructs = ((StructVector) schemaRoot.getVector(name));
+                    if (tuple.getField(name).getValue() != null) {
+                        dateTimeStructs.setIndexDefined(index);
+                        DateDayVector subVectorDay = (DateDayVector) dateTimeStructs.getVectorById(0);
+                        TimeSecVector subVectorTime = (TimeSecVector) dateTimeStructs.getVectorById(1);
+                        LocalDateTime value = (LocalDateTime) tuple.getField(name).getValue();
+                        subVectorDay.setSafe(index, (int) value.toLocalDate().toEpochDay());
+                        subVectorTime.setSafe(index, value.toLocalTime().toSecondOfDay());
+                    }
+                    else dateTimeStructs.setNull(index);
+                    break;
+                case LIST:
+                    // For now only supporting span.
+                    if (((ImmutableList) tuple.getField(name).getValue()).get(0).getClass() != Span.class) {
+                        schemaRoot.clear();
+                        closeClientAndServer();
+                        throw (new DataflowException("Unsupported Element Type for List Field!"));
+                    }
+                    else {
+                        ListVector listVector = (ListVector) schemaRoot.getVector(name);
+                        ImmutableList<Span> spansList = (ImmutableList<Span>) tuple.getField(name).getValue();
+                        try {
+                            convertListOfSpans(spansList, listVector, index, name);
+                        } catch (JsonProcessingException e) {
+                            schemaRoot.clear();
+                            closeClientAndServer();
+                            throw new DataflowException(e.getMessage(), e);
+                        }
+                    }
+
+                    break;
+                default: break;
+            }
+        }
+    }
+
+    /**
+     * For every batch, the operator converts list of {@code Tuple}s into Arrow stream data in almost the exact same
+     * way as it would when using Arrow file, except now it sends stream to the server with
+     * {@link FlightClient#startPut(org.apache.arrow.flight.FlightDescriptor, org.apache.arrow.vector.VectorSchemaRoot,
+     * org.apache.arrow.flight.FlightClient.PutListener, org.apache.arrow.flight.CallOption...)} and {@link
+     * FlightClient.ClientStreamListener#putNext()}. The server uses {@code do_put()} to receive data stream
+     * and convert it into a {@code pyarrow.Table} and store it in the server.
+     * @param values The buffer of tuples to write.
+     */
+    private void writeArrowStream(
+            List<Tuple> values,
+            RootAllocator root,
+            org.apache.arrow.vector.types.pojo.Schema arrowSchema,
+            String descriptorPath) {
+//        System.out.print("Flight Client:\tSending data to Python...");
```

Comment:
remove commendted out code. same applies for the whole PR, there are several places that have commented out code or print statements

Reply:
Will do.

Reply:
However, there are some commented out code that are supposed to be changed back in the future if the pyarrow bug is fixed. Should these codes be deleted?